lib/views/request/show.html.erb

```diff
@@ -6,6 +6,11 @@
 
 <% if flash[:request_sent] %><%= render :partial => 'request_sent' %><% end %>
 
+<%= content_for :ga_pageview do %>
+    window.dataLayer = window.dataLayer || [];
+    ga('require', 'GTM-53XVLCX');
+<% end %>
```

Comment:
This is still going to render to the same place – https://github.com/mysociety/alaveteli/blob/develop/app/views/layouts/default.html.erb#L92

I wonder if the problem here is that `<%=` is rendering – usually you'd use `<%` so that the erb isn't rendered at this point. https://api.rubyonrails.org/v5.0.7/classes/ActionView/Helpers/CaptureHelper.html#method-i-content_for

Reply:
Ah, I just used `<%=` because I was copying the `<%= content_for :javascript do %>` block immediately below. I’ve now tried it with and without the `=` and it doesn’t seem to make a difference to what’s output in the HTML.

BTW, I’m not able to verify your comment that "this is still going to render in the same place". Trying it out locally, when I have the following near the top of `request/show.html.erb`:

```erb
<% content_for :ga_pageview do %>
    window.dataLayer = window.dataLayer || [];
    ga('require', 'GTM-53XVLCX');
<% end %>
<%= render :partial => 'ga_events' %>
```

I get HTML output like this at the end of the request page:

```html
    ga('create', 'UA-24614779-1', 'auto');
            window.dataLayer = window.dataLayer || [];
    ga('require', 'GTM-53XVLCX');
    ga('send', 'pageview', '/request/the_cost_of_boring_two');
```

(Note, `pageview` at the end.)

Whereas, when I render the `ga_events` partial first, like so:

```erb
<%= render :partial => 'ga_events' %>
<% content_for :ga_pageview do %>
    window.dataLayer = window.dataLayer || [];
    ga('require', 'GTM-53XVLCX');
<% end %>
```

I get:

```html
    ga('create', 'UA-24614779-1', 'auto');
            ga('send', 'pageview', '/request/the_cost_of_boring_two');
    window.dataLayer = window.dataLayer || [];
    ga('require', 'GTM-53XVLCX');
```

This makes sense, since [the docs you linked to](https://api.rubyonrails.org/v5.0.7/classes/ActionView/Helpers/CaptureHelper.html#method-i-content_for) confirm that:

> `#content_for` concatenates (default) the blocks it is given for a particular identifier in order.

So the order in which `<%= render :partial => 'ga_events' %>` (and the `<% content_for :ga_pageview do %>` it contains) and our newly added `<% content_for :ga_pageview do %>` are run makes a difference to the order of the lines in the output HTML.

Reply:
PS, I’ve added a fixup commit that swaps the `<%=` for a `<%` – even though it weirdly seems to make no difference, `<% content_for` is clearly the right way to do it.